korma/src/commonMain/kotlin/com/soywiz/korma/interpolation/Easing.kt

```diff
@@ -2,92 +2,171 @@ package com.soywiz.korma.interpolation
+        val EASE_OUT_ELASTIC: Easing get() = Easings.EASE_OUT_ELASTIC
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_BOUNCE"))
+        val EASE_OUT_BOUNCE: Easing get() = Easings.EASE_OUT_BOUNCE
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.LINEAR"))
+        val LINEAR: Easing get() = Easings.LINEAR
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN"))
+        val EASE_IN: Easing get() = Easings.EASE_IN
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT"))
+        val EASE_OUT: Easing get() = Easings.EASE_OUT
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT"))
+        val EASE_IN_OUT: Easing get() = Easings.EASE_IN_OUT
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN"))
+        val EASE_OUT_IN: Easing get() = Easings.EASE_OUT_IN
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_BACK"))
+        val EASE_IN_BACK: Easing get() = Easings.EASE_IN_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_BACK"))
+        val EASE_OUT_BACK: Easing get() = Easings.EASE_OUT_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_BACK"))
+        val EASE_IN_OUT_BACK: Easing get() = Easings.EASE_IN_OUT_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN_BACK"))
+        val EASE_OUT_IN_BACK: Easing get() = Easings.EASE_OUT_IN_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_ELASTIC"))
+        val EASE_IN_OUT_ELASTIC: Easing get() = Easings.EASE_IN_OUT_ELASTIC
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN_ELASTIC"))
+        val EASE_OUT_IN_ELASTIC: Easing get() = Easings.EASE_OUT_IN_ELASTIC
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_BOUNCE"))
+        val EASE_IN_BOUNCE: Easing get() = Easings.EASE_IN_BOUNCE
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_BOUNCE"))
+        val EASE_IN_OUT_BOUNCE: Easing get() = Easings.EASE_IN_OUT_BOUNCE
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN_BOUNCE"))
+        val EASE_OUT_IN_BOUNCE: Easing get() = Easings.EASE_OUT_IN_BOUNCE
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_QUAD"))
+        val EASE_IN_QUAD: Easing get() = Easings.EASE_IN_QUAD
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_QUAD"))
+        val EASE_OUT_QUAD: Easing get() = Easings.EASE_OUT_QUAD
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_QUAD"))
+        val EASE_IN_OUT_QUAD: Easing get() = Easings.EASE_IN_OUT_QUAD
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_SINE"))
+        val EASE_SINE: Easing get() = Easings.EASE_SINE
```

Comment:
The idea of korge-next is to not have `@Deprecated`.

Initially I had `Easing` and `Easings` as public classes (though Easings was not an enum).
Due to discovery and intuition of the users, I tried to merge them in a single class, but it was not possible due to strange issues on the non-JVM targets.
So in the end I kept two classes, and used getters to get it working.
Having the easings as part of the companion object of Easing, make it easier for people that need an `Easing` instance, to know where to get one.

I think it is a good idea what you proposed to be able to iterate over all the standard provided easings, but I would keep this companion object properties and maybe hiding the enum class. I know it is not fully DRY, but at least is statically typed and the implementation happens only once.

Even if the `Easings` enum is private, you can here put something like:

```
val ALL by lazy { Easings.values() }
// or
val ALL_STANDARD by lazy { Easings.values().associateBy { it.name } }
```

This way we are only exposing `Easing` making it simpler to the users, while also supporting iterating the available standard ones and preventing initialization order issues.

What do you think?

Reply:
I definitely agree that `korge-next`, because it's a new major version, shouldn't have deprecated stuff; that should go in the 1.x version, then people will see them before they upgrade to 2.x.

What you suggest sounds reasonable; I'll make that change. And I like the `ALL_STANDARD` variant.